### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by ktonon**

### CODE REVIEW

File: src/AWS/Decode.elm [hunk 1]
```diff
@@ -83,3 +81,45 @@ optional fields decoder =
...
+
+-- dict and helpers
+
+
+dict : JD.Decoder a -> JD.Decoder (Dict String a)
+dict valueDecoder =
+    [ JD.dict valueDecoder
+    , dictAsList valueDecoder
+    ]
+        |> JD.oneOf
```
Comment (ktonon) [inline src/AWS/Decode.elm:103]: Using `oneOf` here to cover multiple possible representations of key values pairs in json. The default dict decoder just assumes a json object. The `dictAsList` is what `getQueueAttributes` actually uses

File: tests/DecodeTests.elm [hunk 2]
```diff
@@ -40,3 +45,80 @@ optionalTests =
...
+    "GetQueueAttributesResult": {
+      "Attributes": [
+        {
+          "Name": "QueueArn",
+          "Value": "arn:aws:sqs:us-east-1:1234567890:elm-test-queue"
+        },
+        {
+          "Name": "ApproximateNumberOfMessages",
+          "Value": "0"
+        },
```
Comment (ktonon) [inline tests/DecodeTests.elm:93]: and this is what it looks like

File: src/AWS/Signers/V4.elm [hunk 3]
```diff
@@ -42,8 +42,7 @@ algorithm =
 
 headers : AWS.Config.Service -> List ( String, String )
 headers config =
-    [ ( "Host", AWS.Http.host config.endpoint )
-    , ( "Accept", "application/json" )
+    [ ( "Accept", "application/json" )
```
Comment (ktonon) [inline src/AWS/Signers/V4.elm:45]: removing host from the list of headers that we try to set. Fixes the error we were seeing in the browser log, but breaks signing so...

File: src/AWS/Signers/V4.elm [hunk 4]
```diff
@@ -79,7 +78,13 @@ addAuthorization :
     -> List ( String, String )
 addAuthorization config creds date req headers =
     [ ( "X-Amz-Date", formatDate date )
-    , ( "Authorization", authorization creds date config req headers )
+    , ( "Authorization"
+      , authorization creds
+            date
+            config
+            req
+            (headers |> (::) ( "Host", AWS.Http.host config.endpoint ))
```
Comment (ktonon) [inline src/AWS/Signers/V4.elm:86]: we just have to add it to the headers used for signing

Comment (ktonon) [reply src/AWS/Signers/V4.elm:86]: i guess this is a funny way to use `::` :|

Comment (nickmonad) [reply src/AWS/Signers/V4.elm:86]: I like it xD

**Review by nickmonad — APPROVED** [review]
+1

**Pull request merged**